test/cert_utils.go

```diff
@@ -145,22 +145,25 @@ func GenerateLeafCertWithExpiration(subject string, oidcIssuer string, expiratio
 	return cert, nil
 }
 
-func GenerateLeafCert(subject string, oidcIssuer string, parentTemplate *x509.Certificate, parentPriv crypto.Signer) (*x509.Certificate, *ecdsa.PrivateKey, error) {
-	certTemplate := &x509.Certificate{
-		SerialNumber:   big.NewInt(1),
-		EmailAddresses: []string{subject},
-		NotBefore:      time.Now().Add(-1 * time.Minute),
-		NotAfter:       time.Now().Add(time.Hour),
-		KeyUsage:       x509.KeyUsageDigitalSignature,
-		ExtKeyUsage:    []x509.ExtKeyUsage{x509.ExtKeyUsageCodeSigning},
-		IsCA:           false,
-		ExtraExtensions: []pkix.Extension{{
+func GenerateLeafCert(subject string, oidcIssuer string, parentTemplate *x509.Certificate, parentPriv crypto.Signer, eexts ...pkix.Extension) (*x509.Certificate, *ecdsa.PrivateKey, error) {
```

Comment:
What do you think about renaming `eexts` to `exts` and then making the old `exts` anonymous?

```
exts = append(exts, []pkix.Extension{ ...
}
```

I could go either way, but  the name `eexts`  took me a little while to understand.

Reply:
that makes a lot of sense ! 🫶